libs/ml/src/ops/reshape.cpp

```diff
@@ -124,14 +124,17 @@ std::vector<math::SizeType> Reshape<TensorType>::ComputeOutputShape(
 }
 
 template <typename TensorType>
-OperationsCount Reshape<TensorType>::ChargeForward() const
+std::pair<OperationsCount, math::SizeVector> Reshape<TensorType>::ChargeForward(
+    std::vector<math::SizeVector> const &input_shapes)
 {
-  assert(!this->batch_output_shape_.empty());
-  OperationsCount cost = fetch::ml::charge_estimation::ops::RESHAPE_PER_ELEMENT *
-                             this->TotalElementsIn({this->batch_output_shape_}) +
-                         fetch::ml::charge_estimation::ops::ASSIGN_PER_ELEMENT *
-                             this->TotalElementsIn({this->batch_output_shape_});
-  return cost;
+  assert(!this->batch_input_shapes_.empty());
+
+  OperationsCount op_cnt = (fetch::ml::charge_estimation::ops::RESHAPE_PER_ELEMENT +
+                            fetch::ml::charge_estimation::ops::RESHAPE_PER_ELEMENT) *
```

Comment:
This should be ASSIGN_PER_ELEMENT?

Reply:
I haven't fine-tuned it. So either is fine at this stage.

Reply:
Okay, I just noticed it had changed from the previous implementation!